app/src/lib/git/core.ts

```diff
@@ -182,9 +182,14 @@ function getDescriptionForError(error: DugiteError): string {
     case DugiteError.SSHAuthenticationFailed:
     case DugiteError.SSHPermissionDenied:
     case DugiteError.HTTPSAuthenticationFailed:
-      return `Authentication failed. You may not have permission to access the repository or the repository may have been archived. Open ${
-        __DARWIN__ ? 'preferences' : 'options'
-      } and verify that you're signed in with an account that has permission to access this repository.`
+      return `Authentication failed. Some common reasons include:
+- You are not logged in to your account: see ${
+        __DARWIN__ ? 'Preferences' : 'Options'
+      }
+- You may need to logout and login to refresh your token.
+- You do not have permission to access this repository.
+- The repository is archived. Check the repository settings on GitHub.com
```

Comment:
I worry about always saying `GitHub.com` here for situations that might not be related to a GitHub repository, such as GitHub Enterprise or a non-GitHub repository.

Could we generalize this to avoid mentioning this website? I've not got strong feels on this, and am open to different wording here, but this is what I'm thinking right now...

```suggestion
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
```